System/Process/Common.hs

```diff
@@ -0,0 +1,215 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE RecordWildCards #-}
+module System.Process.Common
+    ( CreateProcess (..)
+    , CmdSpec (..)
+    , StdStream (..)
+    , ProcessHandle(..)
+    , ProcessHandle__(..)
+    , withFilePathException
+    , PHANDLE
+    , modifyProcessHandle
+    , withProcessHandle
+    , fd_stdin
+    , fd_stdout
+    , fd_stderr
+    , mbFd
+    , mbPipe
+    , pfdToHandle
+
+-- Avoid a warning on Windows
+#ifdef WINDOWS
+    , CGid (..)
```

Comment:
The only comment I really have is still about this. I have read the emails, and I think you're primary motive for doing it like this is so the APIs for Windows and Linux don't diverge. But don't they already?

The `CGid` for Linux seems to be a  type alias (I don't know where it comes from so can't check). I think by exporting the constructor on Windows it's actually giving the opposite impression. To be, without knowing the back story here it looks like it is *actually* being used on windows.

If you do like it this way better, consider Simon's suggestion of adding a `{-# WARNING #-}` so that if someone does use it, they get a warning.